components/global/core/io.cellery.observability.k8s.client/src/main/java/io/cellery/observability/k8s/client/ComponentsEventSource.java

```diff
@@ -0,0 +1,259 @@
+                                " creationTimestamp long, lastKnownActiveTimestamp long, Ingress_Types string" +
+                                ", action string)",
+
+                        description = "This will listen for kubernetes cell events and emit events upon changes " +
+                                "to the pods"
+                )
+        }
+)
+public class ComponentsEventSource extends Source {
+
+    private static final Logger logger = Logger.getLogger(ComponentsEventSource.class.getName());
+
+    private static final String CELL_CRD_GROUP = "mesh.cellery.io";
+    private static final String CELL_CRD_NAME = "cells." + CELL_CRD_GROUP;
+    private static final String INGRESS_TYPES = "Ingress_Types";
+    private static final String HTTP = "HTTP";
+    private static final String WEB = "Web";
+    private static final String TCP = "TCP";
+    private static final String VERSION = "/v1alpha1";
+    private static final String DEFAULT_NAMESPACE = "default";
+    private static final String CELL = "Cell";
+    private static final String LAST_KNOWN_ACTIVE_TIMESTAMP = "lastKnownActiveTimestamp";
+    private KubernetesClient k8sClient;
+    private SourceEventListener sourceEventListener;
+
+    @Override
+    public void init(SourceEventListener sourceEventListener, OptionHolder optionHolder, String[] strings,
+                     ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
+        this.sourceEventListener = sourceEventListener;
+    }
+
+    @Override
+    public Class[] getOutputEventClasses() {
+        return new Class[0];
+    }
+
+    @Override
+    public void connect(ConnectionCallback connectionCallback) throws ConnectionUnavailableException {
+        // Initializing the K8s API Client
+        k8sClient = K8sClientHolder.getK8sClient();
+        if (logger.isDebugEnabled()) {
+            logger.debug("Retrieved API server client instance");
+        }
+
+        // Get CRD for cell resource
+        CustomResourceDefinitionList crdList = k8sClient.customResourceDefinitions().list();
+        List<CustomResourceDefinition> crdsItems = crdList.getItems();
+        CustomResourceDefinition cellCRD = null;
+        for (CustomResourceDefinition crd : crdsItems) {
+            ObjectMeta metadata = crd.getMetadata();
+            if (metadata != null) {
+                String name = metadata.getName();
+                if (CELL_CRD_NAME.equals(name)) {
+                    cellCRD = crd;
+                }
+            }
+        }
+        // Register the custom resource kind cell to Kubernetes deserializer to perform deserialization of cell objects.
+        KubernetesDeserializer.registerCustomKind(CELL_CRD_GROUP + VERSION, CELL, Cell.class);
+
+        // Create client for cell resource
+        NonNamespaceOperation<Cell, CellList, DoneableCell, Resource<Cell, DoneableCell>> cellClient =
```

Comment:
The variable name and the actual stored value does not match.

Suggestion :- Store the client in the variable here and call inNamespace() when you add the watcher.